experiment_pages/experiment/map_rfid.py

```diff
@@ -11,142 +11,257 @@
+                return
+
+            rfid = get_random_rfid()
+            print(f"Generated RFID: {rfid}")
+
+            # Schedule the addition using after() to avoid race conditions
+            self.master.after(0, lambda: self.add_value(rfid))
+            print(f"RFID {rfid} added successfully for animal {len(self.animals)}")
+
+            # Play audio if enabled, with a small delay to avoid threading issues
+            if play_sound:
+                self.master.after(100, lambda: AudioManager.play("shared/sounds/rfid_success.wav"))
+
+        except Exception as e:
+            print(f"Error in add_random_rfid: {e}")
+
+
+
+
+
+    def simulate_all_rfid(self):
+        '''Simulates RFID values for all unmapped animals with increased delay.'''
+        try:
+            unmapped_count = self.db.get_number_animals() - len(self.animals)
+
+            if unmapped_count <= 0:
+                self.raise_warning("All animals are already mapped.")
+                return
+
+            # Use a larger delay between each operation to prevent overload
+            for i in range(unmapped_count):
+                self.master.after(i * 100, self.add_random_rfid, False)  # 100ms delay per animal
+
+            print(f"Simulated {unmapped_count} animals.")
+
+            # Show success message after all RFIDs are added
+            self.master.after(unmapped_count * 100, lambda: messagebox.showinfo(
+                "RFID Simulation Complete",
+                f"{unmapped_count} animals have been successfully mapped with RFID."
+            ))
+
+        except Exception as e:
+            print(f"Error in simulate_all_rfid: {e}")
+
+
+
+    def show_feedback_message(self, unmapped_count):
+        '''Displays a feedback message after RFID simulation.'''
+        try:
+            # Create a message box without the parent argument
+            CTkMessagebox(
+                title="RFID Simulation Complete",
+                message=f"{unmapped_count} animals have been successfully mapped with RFID.",
+                icon="check"
+            )
+        except Exception as e:
+            print(f"Error displaying message box: {e}")
+
+    def setup_navigation_buttons(self, previous_page):
+        '''Set up buttons for serial port selection, change RFID, and removing selections.'''
+
+        # Serial Port Button
+        self.serial_port_button = CTkButton(
+            self, text="Select Serial Port", compound=TOP, width=15,
+            command=self.open_serial_port_selection
+        )
+        self.serial_port_button.place(relx=0.10, rely=0.95, anchor=CENTER)
+
+        # Change RFID Button
+        self.change_rfid_button = CTkButton(
+            self, text="Change RFID", compound=TOP, width=15,
+            command=self.open_change_rfid
+        )
+        self.change_rfid_button.place(relx=0.40, rely=0.95, anchor=CENTER)
+
+        # Delete Button
+        self.delete_button = CTkButton(
```

Comment:
Is this function needed or can we just deal with it when we create the "sacrifice" button. It doesn't remove it from the database so I see no reason to have the button.